create.ps1

```diff
@@ -1,35 +1,190 @@
+                Write-Verbose "SplatParams: $($exportCsvSplatParams | ConvertTo-Json)"
+
+                $updatedCsv = $updatedCsvContent | Foreach-Object { $_ } | Export-Csv @exportCsvSplatParams
+
+                #region Set AccountReference
+                $outputContext.AccountReference = "$($correlationValue)"
+                #endregion Set AccountReference
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "Created row in CSV [$($exportCsvSplatParams.Path)] where [$($correlationField)] = [$($correlationValue)] with AccountReference: $($outputContext.AccountReference | ConvertTo-Json)."
+                        IsError = $false
+                    })
+            }
+            else {
+                Write-Warning "DryRun: Would create row in CSV [$($exportCsvSplatParams.Path)] where [$($correlationField)] = [$($correlationValue)]."
+            }
+            #endregion Create csv row
+
+            break
+        }
+
+        "Correlate" {
+            #region Correlate account
+            $actionMessage = "correlating to CSV row"
+
+            $outputContext.AccountReference = "$($correlationValue)"
+            $outputContext.Data = $currentRow
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    Action  = "CorrelateAccount" # Optionally specify a different action for this audit log
+                    Message = "Correlated to CSV row with AccountReference: $($outputContext.AccountReference | ConvertTo-Json) on [$($correlationField)] = [$($correlationValue)]."
+                    IsError = $false
+                })
+
+            $outputContext.AccountCorrelated = $true
+            #endregion Correlate account
+
+            break
+        }
+
+        "MultipleFound" {
+            #region Multiple accounts found
+            $actionMessage = "correlating to CSV row"
+
+            # Throw terminal error
+            throw "Multiple CSV rows found where [$($correlationField)] = [$($correlationValue)]. Please correct this so the persons are unique."
+            #endregion Multiple accounts found
+
+            break
+        }
+    }
+    #endregion Process
+    #endregion Account
 }
-finally{
-	$mutex.ReleaseMutex()
+catch {
+    $ex = $PSItem
+    $auditMessage = "Error $($actionMessage). Error: $($ex.Exception.Message)"
+    Write-Warning "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+
+    $outputContext.AuditLogs.Add([PSCustomObject]@{
+            # Action  = "" # Optional
+            Message = $auditMessage
+            IsError = $true
+        })
 }
+finally {
+    # Check if auditLogs contains errors, if no errors are found, set success to true
+    if ($outputContext.AuditLogs.IsError -contains $true) {
+        $outputContext.Success = $false
+    }
+    else {
+        $outputContext.Success = $true
+    }
```

Comment:
Technically we could move this to the try and catch block. Which eliminates the use of the finally block.

Reply:
This is true but this csv connector is a template for a more complex connector, where this functionality might be required